bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/ResumingStreamingResultScanner.java

```diff
@@ -94,27 +95,48 @@ public Row next() throws IOException {
         if (result != null) {
           lastRowKey = result.getKey();
           rowCount++;
-          // We've had at least one successful RPC, reset the backoff
-          currentBackoff = null;
+          // We've had at least one successful RPC, reset the backoffs
+          currentErrorBackoff = null;
+          currentTimeoutBackoff = null;
         }
 
         return result;
       } catch (ScanTimeoutException rte) {
-        logger.info("The client could not get a response in %d ms. Retrying the scan.",
-          retryOptions.getReadPartialRowTimeoutMillis());
-        backOffAndRetry(rte);
+        handleScanTimeout(rte);
       } catch (IOExceptionWithStatus ioe) {
-        Status.Code code = ioe.getStatus().getCode();
-        if (retryOptions.isRetryable(code)) {
-          logger.info("Reissuing scan after receiving error with status: %s.", code.name());
-          backOffAndRetry(ioe);
-        } else {
-          throw ioe;
-        }
+        handleIOException(ioe);
       }
     }
   }
 
+  private void handleScanTimeout(ScanTimeoutException rte) throws IOException {
+    logger.info("The client could not get a response in %d ms. Retrying the scan.",
+      retryOptions.getReadPartialRowTimeoutMillis());
+    if (currentTimeoutBackoff == null) {
+      // Create a backoff that retries immediately, up to the max number of times
+      currentTimeoutBackoff =
+          new TimeoutRetryBackOff(retryOptions.getMaxScanTimeoutRetries(), BackOff.ZERO_BACKOFF);
```

Comment:
The alternative is to return 0 explicitly in the TimeoutRetryBackOff implementation. I thought this provided an easier path to doing something fancier (like exponential backoff) instead of immediate retries in the future. Happy to remove if you think the extra code isn't worth it.